src/memory-analyzer/analyze_symbol.cpp

```diff
@@ -315,7 +316,16 @@ exprt symbol_analyzert::get_pointer_value(
   {
     if(is_c_char(expr.type().subtype()))
     {
-      return get_char_pointer_value(expr, memory_location, location);
+      INVARIANT(
+        !points_to_member(value),
+        "pointers-to-char shouldn't point to members");
+      const auto result_pointee_expr =
+        get_char_pointer_value(expr, memory_location, location);
+      const auto maybe_cast_result = typecast_exprt::conditional_cast(
```

Comment:
hmm. My guess is you had to understand something here about memory analyser. 
Any clarification would greatly help reviewers. This code was inherited, but as we understand it (to extend it) - we should clarify, so as to not duplicate effort when this code is passed on to others.

Reply:
Done, or maybe rewritten.